src/reason-parser/reason_pprint_ast.ml

```diff
@@ -1759,9 +1765,11 @@ let consolidateSeparator = preOrderWalk (function
        when listConfig.sep <> ""
          && listConfig.sepLeft
     ->
+     (* TODO: (perf) cache List.length *)
      let layoutsWithSepAndComment =
        List.mapi (fun i layout ->
            (* Do not render the final separator *)
+           (* TODO: Support !sepLeft, and this should apply to the *first* separator if !sepLeft.  *)
```

Comment:
Thought I'd leave a couple of notes about some quadradic running times I've found while refactoring.